-------------------------------------------------------------------------------
Test set: com.ims.services.UserServiceTest
-------------------------------------------------------------------------------
Tests run: 12, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.133 s <<< FAILURE! -- in com.ims.services.UserServiceTest
com.ims.services.UserServiceTest.deleteUser_WithNonExistentUser_ShouldThrowEntityNotFoundException -- Time elapsed: 0.002 s <<< FAILURE!
org.opentest4j.AssertionFailedError: Unexpected exception type thrown, expected: <jakarta.persistence.EntityNotFoundException> but was: <java.lang.RuntimeException>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:67)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:35)
	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3115)
	at com.ims.services.UserServiceTest.deleteUser_WithNonExistentUser_ShouldThrowEntityNotFoundException(UserServiceTest.java:331)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.RuntimeException: Failed to delete user
	at com.ims.services.UserService.deleteUser(UserService.java:111)
	at com.ims.services.UserServiceTest.lambda$deleteUser_WithNonExistentUser_ShouldThrowEntityNotFoundException$6(UserServiceTest.java:333)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:53)
	... 6 more
Caused by: jakarta.persistence.EntityNotFoundException: User not found with ID: 1
	at com.ims.services.UserService.deleteUser(UserService.java:100)
	... 8 more

